aiida_cusp/parsers/vasp_file_parser.py

```diff
@@ -71,15 +71,30 @@ def parse(self, **kwargs):
         exit_code = self.register_output_nodes(parsed_results)
         return exit_code
 
-    def normalized_filename(self, filepath):
+    def normalized_filename(self, name):
         """
         Return a normalized version of the filename, i.e. lower case and
         .suffix replace with _suffix
         """
-        return str(filepath.name.lower()).replace(".", "_")
+        return str(name.lower()).replace(".", "_")
+
+    def split_suffix(self, name):
+        """
+        Split the filename and return the prefix and suffix separately
+        """
+        exist_parsers = ["generic", "wavecar", "chgcar",
+                         "contcar", "vasprun.xml", "outcar"]
+        formal_name = name
+        suffix = ""
+        for parser in exist_parsers:
+            if name.lower().startswith(parser):
+                formal_name = parser
+                suffix = name[len(parser):]
+        return formal_name, suffix
```

Comment:
I think this function should really only remove the suffix independent of the actual filename. The hook name `parse_{some_filename}` is build for all files. If a hook for that specific file is available or if the `parse_generic` hook needs to be used is then checked later in the parsing process. I'm afraid, to make that work with suffixes we need to change the complete parsing process.

Reply:
hmmm. It's a bit hard to distinguish what is a manually added suffix by custodian and what is an actual file type suffix `.xml` . Is there anyway that we get the job list info in the parser class?

Reply:
alternatively, I can enumerate all the class methods from the parser to obtain the `exist_parsers` list. Would that work?

Reply:
Actually, the `exist_parsers` list is not necessary (as I said, the hook name is generated for all filenames, it doesn’t care if the hook exists or not. This is checked later in der `parse()` function, which will call the generic parsing hook if the defined one is not available) But you are absolutely right in that we need to distinguish the suffixes. So I think, we need to add some more logic to the parser. I‘m a bit busy right now but I will have a closer look at this next week. (BTW it would be good if you could move your changes to a separate branch instead of develop directly ;) )